integration/airflow/openlineage/airflow/extractors/python_extractor.py

```diff
@@ -35,27 +40,61 @@ def extract(self) -> Optional[TaskMetadata]:
                 "sourceCode": SourceCodeJobFacet(
                     "python",
                     # We're on worker and should have access to DAG files
-                    source_code
+                    source_code,
                 )
             }
+        """
+        This allows for the collection of manual lineage data. When OPENLINEAGE_COLLECT_MANUALLY
+        is set to true, PythonExtractor will extract Datasets, when set to false,
+        inputs and outputs are set to empty Lists.
+        """
+        collect_manual_lineage = False
+        if os.environ.get("OPENLINEAGE_COLLECT_MANUALLY", "False").lower() in (
```

Comment:
I'd prefer we fallback on using `inlets`/`outlets` rather than having the lineage extractor handle _only_ manual extraction or _only_ automated extraction. That is, there are some cases where lineage extraction for an operator is not yet supported, or for some reason the extraction / SQL parsing failed. It'd be great to fallback to the user-defined `inlets`/`outlets` (if defined) to send along with the OpenLineage event.

**The order of precedence would be:**
1. Use extractor
2. Use `inlets`/`outlets` (if defined)

Now, there will be the case where both the extractor and `inlets`/`outlets` are defined. In that case, I would go with the result from the extractor as "the code" is a more complete representation of "reality" than `inlets`/`outlets` that are user-defined and may not have been updated in a while. In this case, I would log a warning that `inlets`/`outlets` were defined, but the extractor lineage metadata will be used.

/cc @mobuchowski

Reply:
Yeah, I wonder if this just should not be an extractor, but a special case in `listener`. Or, might be an extractor, but just applied when extractor for specific operator is not found. 

The exception for that is `PythonExtractor` and `BashExtractor` which exist solely to provide additional debug information, not lineage. So we'd need to reconcile both.

Reply:
@mobuchowski: We'd want the `inlet`/`outlet` extraction to be the default for all extractor that extend [`BaseExtractor`](https://github.com/OpenLineage/OpenLineage/blob/main/integration/airflow/openlineage/airflow/extractors/base.py). This would include support for the `PythonExtractor` and `BashExtractor` where I think annotated lineage would be the most useful (the Airflow [lineage docs](https://airflow.apache.org/docs/apache-airflow/stable/lineage.html) outline usage for `BashOperator`).

It would be great to merge this PR, but I think we have the following open questions:
1. How do we reconcile extraction of lineage via `inlet`/`outlet`? But, also can we merge this PR to provide initial support, then introduce a more generic approach later?
2. How should `inlet`/`outlet` be defined in order to extract metadata about the dataset (`schema`, `source`, etc)?